verification/challengeresponse_test.go

```diff
@@ -759,3 +799,182 @@ func TestChallengeResponseConfig_Run_async_with_explicit_delete_failed(t *testin
 	assert.EqualError(t, err, "session resource in failed state")
 	assert.Nil(t, result)
 }
+
+func TestChallengeResponseConfig_Run_async_CMWWrap_ok(t *testing.T) {
+	sessionState := []string{`
+{
+    "nonce": "3q2+7w==",
+    "expiry": "2030-10-12T07:20:50.52Z",
+    "accept": [
+        "application/psa-attestation-token"
+    ],
+    "status": "waiting"
+}`, `
+{
+    "nonce": "3q2+7w==",
+    "expiry": "2030-10-12T07:20:50.52Z",
+    "accept": [
+        "application/psa-attestation-token"
+    ],
+    "status": "processing",
+	"evidence": {
+        "type": "application/vnd.veraison.cmw+cbor",
+        "value": "g3giYXBwbGljYXRpb24vbXktZXZpZGVuY2UtbWVkaWEtdHlwZUMODQ4E"
+    }
+}`, `
+{
+    "nonce": "3q2+7w==",
+    "expiry": "2030-10-12T07:20:50.52Z",
+    "accept": [
+        "application/psa-attestation-token"
+    ],
+    "status": "complete",
+	"evidence": {
+        "type": "application/vnd.veraison.cmw+cbor",
+        "value": "g3giYXBwbGljYXRpb24vbXktZXZpZGVuY2UtbWVkaWEtdHlwZUMODQ4E"
+    },
+	"result": {
+        "is_valid": true,
+		"claims": {}
+    }
+}`,
+	}
+
+	iter := 1
+	h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch iter {
+		case 1:
+			assert.Equal(t, http.MethodPost, r.Method)
+
+			w.Header().Set("Location", testRelSessionURI)
+			w.WriteHeader(http.StatusCreated)
+			_, e := w.Write([]byte(sessionState[0]))
+			require.Nil(t, e)
+
+			iter++
+		case 2:
+			assert.Equal(t, http.MethodPost, r.Method)
+			ev, err := ioutil.ReadAll(r.Body)
+			require.Nil(t, err)
+			assert.Equal(t, ev, testCMWEvCBOR)
+			assert.Equal(t, r.Header.Get("Content-Type"), testCMWCtCBOR)
+			w.WriteHeader(http.StatusAccepted)
+			_, e := w.Write([]byte(sessionState[1]))
+			require.Nil(t, e)
+
+			iter++
+		case 3:
+			assert.Equal(t, http.MethodGet, r.Method)
+
+			w.WriteHeader(http.StatusOK)
+			_, e := w.Write([]byte(sessionState[2]))
+			require.Nil(t, e)
+		}
+	})
+
+	client, teardown := common.NewTestingHTTPClient(h)
+	defer teardown()
+
+	cfg := ChallengeResponseConfig{
+		Nonce:           testNonce,
+		NewSessionURI:   testNewSessionURI,
+		EvidenceBuilder: testEvidenceBuilder{},
+		Client:          client,
+		Wrap:            WrapCBOR,
+	}
+	expectedResult := `{ "is_valid": true, "claims": {} }`
+
+	result, err := cfg.Run()
+	assert.NoError(t, err)
+	assert.JSONEq(t, expectedResult, string(result))
+}
+
+func TestChallengeResponseConfig_Run_async_CMWWrap_JSON_ok(t *testing.T) {
```

Comment:
please factor this out (it's the same as CBOR_ok)

Reply:
Will be done as part of issue #29